### PR TITLE
Update stageNumber to lessonNumber

### PR DESCRIPTION
**Pull request by dmcavoy**
Update stageNumber to lessonNumber across code base.

https://codedotorg.atlassian.net/browse/PLAT-981

### CODE REVIEW

File: apps/src/sites/studio/pages/scripts/lesson_extras.js [hunk 1]
```diff
@@ -22,8 +22,8 @@ config.bonusLevels = config.bonusLevels.map(bonus => {
 ReactDOM.render(
   <Provider store={store}>
     <LessonExtras
-      lessonNumber={config.stageNumber}
-      nextLessonNumber={config.nextStageNumber}
+      lessonNumber={config.lessonNumber}
+      nextLessonNumber={config.nextLessonNumber}
```
Comment (davidsbailey) [inline apps/src/sites/studio/pages/scripts/lesson_extras.js:26]: looks like you would need to update this codepath also:
* https://github.com/code-dot-org/code-dot-org/blob/842dc7e4d76a70531c44a0f372e39b1c1e5f5f08/dashboard/app/controllers/script_levels_controller.rb#L272-L274
* https://github.com/code-dot-org/code-dot-org/blob/503d587324b19cfe9993c316e0166de1db9f2c32/dashboard/app/views/scripts/lesson_extras.html.haml#L7

File: dashboard/app/controllers/script_levels_controller.rb [hunk 2]
```diff
@@ -270,7 +270,7 @@ def stage_extras
     end
 
     @stage_extras = {
-      next_stage_number: @lesson.next_level_number_for_lesson_extras(user),
+      next_lesson_number: @lesson.next_level_number_for_lesson_extras(user),
       stage_number: @lesson.relative_position,
```
Comment (davidsbailey) [inline dashboard/app/controllers/script_levels_controller.rb]: sorry I missed that the file I was trying to comment on is already in this PR. I could be misreading, but it looks like you  might need to update stage_number here too for this to not break

Comment (dmcavoy) [reply dashboard/app/controllers/script_levels_controller.rb]: good catch. I updated it.

File: apps/src/sites/studio/pages/scripts/lesson_extras.js (hunk 1 above)
Comment (dmcavoy) [reply apps/src/sites/studio/pages/scripts/lesson_extras.js:26]: Thanks for your thoughtful review. I think this is fixed now

**Review by davidsbailey — APPROVED** [review]

**Pull request merged**